src/core_plugins/getting_started/public/lib/add_setup_work.js

```diff
@@ -68,14 +67,14 @@ uiRoutes
         });
         notify.error('Please create a new index pattern');
         kbnUrl.change(CREATE_INDEX_PATTERN_ROUTE);
-        return;
+        throw WAIT_FOR_URL_CHANGE_TOKEN;
```

Comment:
@ycombinator I tried a different approach, throwing this token from within a route setup work function will halt the setup work, allowing it to wait for `kbnUrl` to change. What do you think?

/cc @cjcenizal